src/edit.c

```diff
@@ -18,15 +18,41 @@ void get_edits(const char *p_row, const char *q_row,
     assert(p && q && edits); // Don't want to handle allocation failures
 
     // FIXME: do the actual calculations here
+    memset(p, 0, n+1); 
+    memset(q, 0, n+1);
+    memset(edits, 0, n+1);
 
-    p[0] = '\0';     // for now we just make
+    int j = 0; // for the p variable
+    int k = 0; // for the q variable
+   for (int i = 0 ; i < n ; i ++) {
+        if (q_row[i] == '-') {
+            edits[i] = 'D';
+            p[j] = p_row[i]; 
+            j++;
+        }
+        else if (p_row[i] == '-') {
+            edits[i] = 'I';
+            q[k] = q_row[i];
+            k++; 
+        }
+        else {
+            edits[i] = 'M';
+            p[j] = p_row[i];
+            q[k] = q_row[i];
+            j++;
+            k++;
+        }
+    }
```

Comment:
The nice thing about the `++` operator in C is that it is an expression as well as a statement. It is particularly nice when you do something like this, where we need to write to a buffer and increment to the next insertion point. It is all too easy to forget to increment the index, but if you always write code such as

```c
    buf[i++] = val;
```

then you never forget. In this code, I think you should combine the assignments and increments to make the code more C-like.